qt and qtbase are broken: linking errors in manually built test

```
# build test the manual way mkdir 
'/usr/lib/mxe/tmp-qt-i686-w64-mingw32.static/qt-everywhere-opensource-
src-4.8.7/test-qt-pkgconfig' 
'/usr/lib/mxe/usr/i686-w64-mingw32.static/qt/bin/uic' -o 
'/usr/lib/mxe/tmp-qt-i686-w64-mingw32.static/qt-everywhere-opensource-
src-4.8.7/test-qt-pkgconfig/ui_qt-test.h' './src/qt-test.ui' 
'i686-w64-mingw32.static-g++' -W -Wall -Werror -std=c++0x -pedantic 
'./src/qt-test.cpp' -o 
'/usr/lib/mxe/usr/i686-w64-mingw32.static/bin/test-qt-pkgconfig.exe' 
-I'/usr/lib/mxe/tmp-qt-i686-w64-mingw32.static/qt-everywhere-opensourc
e-src-4.8.7/test-qt-pkgconfig' `'i686-w64-mingw32.static-pkg-config' 
QtGui --cflags --libs` 
/tmp/ccShJ07f.o:qt-test.cpp:(.text$_ZN10MainWindowC1EP7QWidget[__ZN10M
ainWindowC1EP7QWidget]+0x6e): undefined reference to `vtable for 
MainWindow' 
/tmp/ccShJ07f.o:qt-test.cpp:(.text$_ZN10MainWindowC1EP7QWidget[__ZN10M
ainWindowC1EP7QWidget]+0x78): undefined reference to `vtable for 
MainWindow' 
/tmp/ccShJ07f.o:qt-test.cpp:(.text$_ZN10MainWindowD1Ev[__ZN10MainWindo
wD1Ev]+0xa): undefined reference to `vtable for MainWindow' 
/tmp/ccShJ07f.o:qt-test.cpp:(.text$_ZN10MainWindowD1Ev[__ZN10MainWindo
wD1Ev]+0x14): undefined reference to `vtable for MainWindow' 
collect2: error: ld returned 1 exit status
```

In #851 I added [a header](https://github.com/mxe/mxe/commit/dbe8eef5f5c500758dd3ddafe9e6f94771d7fd97) and [QRC resource](https://github.com/mxe/mxe/commit/6004e1e00affe07ea29e4e073d006ac686a00937) to test MOC and RCC tools, but did not update build instructions for the test. The fix is coming soon.

--- a/src/qt.mk
+++ b/src/qt.mk
@@ -106,9 +106,17 @@ define $(PKG)_BUILD
     # build test the manual way
     mkdir '$(1)/test-$(PKG)-pkgconfig'
     '$(PREFIX)/$(TARGET)/qt/bin/uic' -o '$(1)/test-$(PKG)-pkgconfig/ui_qt-test.h' '$(TOP_DIR)/src/qt-test.ui'
+    '$(PREFIX)/$(TARGET)/qt/bin/moc' \
+        -o '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
+        -I'$(1)/test-$(PKG)-pkgconfig' \
+        '$(TOP_DIR)/src/qt-test.hpp'
+    '$(PREFIX)/$(TARGET)/qt/bin/rcc' -name qt-test -o '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' '$(TOP_DIR)/src/qt-test.qrc'
     '$(TARGET)-g++' \
         -W -Wall -Werror -std=c++0x -pedantic \
-        '$(TOP_DIR)/src/qt-test.cpp' -o '$(PREFIX)/$(TARGET)/bin/test-$(PKG)-pkgconfig.exe' \
+        '$(TOP_DIR)/src/qt-test.cpp' \
+        '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
+        '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' \
+        -o '$(PREFIX)/$(TARGET)/bin/test-$(PKG)-pkgconfig.exe' \
         -I'$(1)/test-$(PKG)-pkgconfig' \
         `'$(TARGET)-pkg-config' QtGui --cflags --libs`
 

--- a/src/qtbase.mk
+++ b/src/qtbase.mk
@@ -72,9 +72,17 @@ define $(PKG)_BUILD
     # build test the manual way
     mkdir '$(1)/test-$(PKG)-pkgconfig'
     '$(PREFIX)/$(TARGET)/qt5/bin/uic' -o '$(1)/test-$(PKG)-pkgconfig/ui_qt-test.h' '$(TOP_DIR)/src/qt-test.ui'
+    '$(PREFIX)/$(TARGET)/qt5/bin/moc' \
+        -o '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
+        -I'$(1)/test-$(PKG)-pkgconfig' \
+        '$(TOP_DIR)/src/qt-test.hpp'
+    '$(PREFIX)/$(TARGET)/qt5/bin/rcc' -name qt-test -o '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' '$(TOP_DIR)/src/qt-test.qrc'
     '$(TARGET)-g++' \
         -W -Wall -Werror -std=c++0x -pedantic \
-        '$(TOP_DIR)/src/qt-test.cpp' -o '$(PREFIX)/$(TARGET)/bin/test-$(PKG)-pkgconfig.exe' \
+        '$(TOP_DIR)/src/qt-test.cpp' \
+        '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
+        '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' \
+        -o '$(PREFIX)/$(TARGET)/bin/test-$(PKG)-pkgconfig.exe' \
         -I'$(1)/test-$(PKG)-pkgconfig' \
         `'$(TARGET)-pkg-config' Qt5Widgets --cflags --libs`
 

